salt/states/file.py

```diff
@@ -6887,6 +6887,12 @@ def _patch(patch_file, options=None, dry_run=False):
             patch_opts.append("-p{}".format(strip))
 
         pre_check = _patch(patch_file, patch_opts)
+
+        if os.path.getsize(patch_rejects) == 0 and pre_check['retcode'] != 0:
+            ret['comment'] = pre_check['stderr']
+            ret['result'] = False
+            return ret
+
         if pre_check["retcode"] != 0:
             # Try to reverse-apply hunks from rejects file using a dry-run.
```

Comment:
Just to be super-safe, I would suggest the following change. Move your new code inside the ``if pre_check["retcode"] != 0:`` conditional, immediately after this line. Then change your conditional to be ``if not os.path.exists(patch_rejects) or os.path.getsize(patch_rejects) == 0``. If the user specifies ``reject_file`` then I do not think this file will necessarily exist. The extra ``os.path.exists()`` check will ensure this doesn't result in ``OSError``.

Reply:
@danielrobbins fixed. I was trying to provide unit test but I'm lack of knowledge how to mock content of `salt://example.patch` in order to run test for `file.patch`. Please verify if current state is enough or do I need to write a proper unit test.